.github/workflows/health.yaml

```diff
@@ -10,5 +10,6 @@ jobs:
       coverage_web: false
       checks: "version,changelog,license,do-not-submit,breaking"
       use-flutter: true
+      sdk: dev
```

Comment:
Does it work if you put an actual version number here instead of "dev"?

Reply:
I think the health check is running as intended, only the publish validation workflow needed the dev version. The health check fails due to a missing license on a golden file which never had a license header (and on the libclang bindings file, which probably shouldn't have a Dart license header).

Reply:
Right, that should not have a header file.

Though I'm still seeing Dart versio issues on the CI?

```
Done, found 52 files without license headers
Collecting packages without changed changelogs:
Done, found 5 packages.
Collecting files without license headers in those packages:
Done, found 0 packages with a need for a changelog.
Look for changes in pkgs/ffigen with base ../../../base_repo/pkgs/ffigen
Error: Exception: The current Dart SDK version is 3.2.3.

Because ffigen requires SDK version >=3.3.0-252.0.dev <4.0.0, version solving failed.


You can try the following suggestion to make the pubspec resolve:
* Try using the Dart SDK version: 3.3.0. See https://dart.dev/get-dart.

[...]

Changed 55 dependencies!
1 package has newer versions incompatible with dependency constraints.
Try `dart pub outdated` for more information.
Preparing .
Copying sources from .
Preparing package dependencies for local package .
Resolving dependencies...
The current Dart SDK version is 3.2.3.

Because ffigen requires SDK version >=3.3.0-252.0.dev <4.0.0, version solving failed.


You can try the following suggestion to make the pubspec resolve:
* Try using the Dart SDK version: 3.3.0. See https://dart.dev/get-dart.

Unhandled exception:
PathNotFoundException: Cannot open file, path = 'pkgs/ffigen/report.json' (OS Error: No such file or directory, errno = 2)
#0      _File.throwIfError (dart:io/file_impl.dart:675:7)
#1      _File.openSync (dart:io/file_impl.dart:490:5)
#2      _File.readAsBytesSync (dart:io/file_impl.dart:574:18)
#3      _File.readAsStringSync (dart:io/file_impl.dart:624:18)
#4      Health.breakingCheck (package:firehose/src/health/health.dart:137:41)
#5      Health.healthCheck (package:firehose/src/health/health.dart:82:30)
<asynchronous suspension>
#6      main (file:///opt/hostedtoolcache/flutter/stable-3.16.5-x64/.pub-cache/hosted/pub.dev/firehose-0.4.2/bin/health.dart:32:3)
<asynchronous suspension>
Error: Process completed with exit code 255.
```

https://github.com/dart-lang/native/actions/runs/7402310627/job/20139919918?pr=860

@mosuem Any guidance on how to make the health check happy?

* We have files in third_party/ folders which require a different license than the Dart one
* In various places the health check fails to resolve the Dart SDK version.

Reply:
I will check it out. Thanks for dogfooding the health workflow :)